### DISCUSSION TITLE
Location of declaration

### DISCUSSION THREAD

**Discussion by carloszimm** [Hookyns/tst-reflect#88] [Q&A] [answered]
Is there any way to inquiry a type (classes, functions, etc) about the file where they are declared or it is only achievable through a parser for instance?

**Comment by Hookyns** [comment, marked as answer]
Hi, 
if getting the constructor by `getCtor()` is not enough, you can enable "server" mode if you want to know exact file locations. Then you can use `type.constructorDescription` property. See [configuration](https://github.com/Hookyns/tst-reflect/wiki/Configuration).

Is this something you want?
```typescript
import { getType, Type, AccessModifier, Accessor } from 'tst-reflect';

function logTypeDescription<T>() {
    console.log(getType<T>().constructorDescription);
}

export class SomeClass {
    property1: string = 'p1';
    property2: string = 'p2';
    property3: number = 3;

    get property4(): 'foo' | 'bar' {
        return 'foo';
    }

    constructor(a: string) {}

    method1(): void {}
}

logTypeDescription<SomeClass>();

// Get Ctor of the type, create instance and log value of "property4"
getType<SomeClass>()
    .getCtor()
    .then((SomeClassCtor) => {
        console.log(new SomeClassCtor().property4);
    });

```

output:
```
ConstructorImportActivator {
  _name: 'SomeClass',
  _exportName: 'SomeClass',
  _sourcePath: '/home/projects/tst-reflect-find-inheritors-pdvjbb/src/index.ts',
  _outPath: '/home/projects/tst-reflect-find-inheritors-pdvjbb/dist/index.js'
}
foo
```

StackBlitz [demo](https://stackblitz.com/edit/tst-reflect-server-mode?file=src%2Findex.ts).

**Reply by carloszimm** [reply]
Yep. It will do the job! Thanks, @Hookyns :smiley:

**Reply by Hookyns** [reply]
May I ask why you need it?

Just yesterday I removed the path from the Alpha version. In the Alpha version there are types `Module` and `Type`. `Type` has only the `getCtor()`. `Module` had property `path` and method `import(): Promise<{ [exportName: string]: any }>`. I removed the path a kept only the import. Will this do the job or you rly need the path?

FS path is useless or even danger for browsers and I want to keep the new version `mode`less, but if you have some use-case I think I have to add some option to enable it.

**Reply by carloszimm** [reply]
Sure. I'm working on a project in which I'm gathering info about declarations according to their module (and since modules in TS/JS are referred to through their [relative] path, generally, I asked about a way to recovery this info). This project will be be pretty much used in the server side (no browser involved).

To be frank, when I asked, I was looking for alternatives just in case it became difficult to accomplish it trough a compiler/parser tool such as ts-morph. Until future discussion with varying use cases (and more contributors), it is probably better to keep your reasoning in the Alpha version.
